fe/fe-core/src/main/java/org/apache/doris/analysis/TimeV2Literal.java

```diff
@@ -0,0 +1,269 @@
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.analysis;
+
+import org.apache.doris.catalog.ScalarType;
+import org.apache.doris.catalog.Type;
+import org.apache.doris.common.FormatOptions;
+import org.apache.doris.nereids.exceptions.AnalysisException;
+import org.apache.doris.thrift.TExprNode;
+import org.apache.doris.thrift.TExprNodeType;
+import org.apache.doris.thrift.TTimeV2Literal;
+
+public class TimeV2Literal extends LiteralExpr {
+
+    public static final TimeV2Literal MIN_TIME = new TimeV2Literal(-838, 0, 0, 0, 0);
+    public static final TimeV2Literal MAX_TIME = new TimeV2Literal(838, 59, 59, 999999, 6);
+
+    protected float hour;
+    protected long minute;
+    protected long second;
+    protected long microsecond;
+
+    /**
+     * C'tor forcing type, e.g., due to implicit cast
+     */
+    // for restore
+    private TimeV2Literal() {
+        this.type = Type.TIMEV2;
+        this.hour = 0;
+        this.minute = 0;
+        this.second = 0;
+        this.microsecond = 0;
+    }
+
+    public TimeV2Literal(float hour, long minute, long second) {
+        super();
+        this.type = Type.TIMEV2;
+        this.hour = hour;
+        this.minute = minute;
+        this.second = second;
+        this.microsecond = 0;
+        analysisDone();
+    }
+
+    public TimeV2Literal(float hour, long minute, long second, long microsecond, long scale) {
+        super();
+        this.type = ScalarType.createTimeV2Type((int) scale);
+        this.hour = hour;
+        this.minute = minute;
+        this.second = second;
+        this.microsecond = microsecond;
+        while (microsecond != 0 && this.microsecond < 100000) {
+            this.microsecond *= 10;
+        }
+        analysisDone();
+    }
+
+    public TimeV2Literal(String s) throws AnalysisException {
+        super();
+        init(s);
+        analysisDone();
+    }
+
+    protected TimeV2Literal(TimeV2Literal other) {
+        super(other);
+        this.type = ScalarType.createTimeV2Type(((ScalarType) other.type).getScalarScale());
+        this.hour = other.getHour();
+        this.minute = other.getMinute();
+        this.second = other.getSecond();
+        this.microsecond = other.getMicroSecond();
+    }
+
+    @Override
+    public Expr clone() {
+        return new TimeV2Literal(this);
+    }
+
+    protected void init(String s) throws AnalysisException {
+        // should like be/src/vec/runtime/time_value.h timev2_to_double_from_str
+        if (!s.contains(":")) {
+            boolean sign = false;
+            String tail = "";
```

Comment:
could u split this function into two functions: normalize string and parse string